src/LinearElastic.jl

```diff
@@ -31,30 +31,27 @@ function elastic_tangent_3D(E::T, ν::T) where T
     return C
 end
 
-# TODO not sure if we could hardcode Float64 here - do we want to differentiate states?
-struct LinearElasticState{dim, T, M} <: AbstractMaterialState
-    σ::SymmetricTensor{2,dim,T,M} # stress
-end
-
-# TODO could these be automatically generated?
-Base.zero(::Type{LinearElasticState{dim,T,M}}) where {dim,T,M} = LinearElasticState(zero(SymmetricTensor{2,dim,T,M}))
+struct LinearElasticState <: AbstractMaterialState end
 
 # define which state belongs to the material
-initial_material_state(::LinearElastic) = zero(LinearElasticState{3,Float64,6})
+initial_material_state(::LinearElastic) = LinearElasticState()
+
+# for restricted stress states to create buffer
+get_stress_type(::LinearElasticState) = SymmetricTensor{2,3,Float64,6}
 
 # constitutive drivers generally operate in 3D 
 # (we could specialize for lower dimensions if needed for performance)
 """
-    material_response(m::LinearElastic, Δε::SymmetricTensor{2,3}, state::LinearElasticState{3})
+    material_response(m::LinearElastic, Δε::SymmetricTensor{2,3})
 
-Return the stress tensor, stress tangent and the new `MaterialState` for the given strain step Δε such that
+Return the stress tensor and the stress tangent for the given strain ε such that
 
 ```math
 \\boldsymbol{\\sigma} = \\mathbf{E}^\\text{e} : \\Delta \\boldsymbol{\\varepsilon} .
 ```
```

Comment:
Remove Δ in function signature and equation?

Reply:
Good catch, thanks!